File: src/main/java/org/droidmate/analyzer/evaluation/DidNotCrash.java
package org.droidmate.analyzer.evaluation;

import org.droidmate.analyzer.exploration.IExplorationResult;

/**
 * Evaluate the scenario on the basis that it did not crash
 */
@SuppressWarnings("unused")
public class DidNotCrash implements IEvaluationStrategy {
    DidNotCrash(){
        super();
    }

    @Override
    public boolean isValid(IExplorationResult result) {
        return !result.hasCrashed();
    }

    @Override
    public double getDissimilarity(IExplorationResult result) {
        if (this.isValid(result))
            return 0;
        else
            return 1;
    }
}

File: src/main/kotlin/org/droidmate/analyzer/evaluation/DidNotCrash.kt
package org.droidmate.analyzer.evaluation

import org.droidmate.analyzer.exploration.IExplorationResult

/**
 * Evaluate the scenario on the basis that it did not crash
 */
open class DidNotCrash internal constructor() : IEvaluationStrategy {

    override fun isValid(result: IExplorationResult): Boolean {
        return !result.hasCrashed()
    }

    override fun getDissimilarity(result: IExplorationResult): Double {
        if (this.isValid(result))
            return 0.0
        else
            return 1.0
    }
}
